### PR TITLE
Filter traffic by status code

### PR DESCRIPTION
**Pull request by andertun**
https://github.com/kapetacom/provider-resource-internal-rest/assets/1045799/18218492-e095-4e55-ae92-56dff9a5f148

### CODE REVIEW

File: src/web/components/table/KapTable.tsx [hunk 1]
```diff
@@ -27,18 +27,42 @@ export const KapTable = <Data extends { id: string }>(props: KapTableProps<Data>
...
         setOrder(isAsc ? 'desc' : 'asc');
         setOrderBy(property);
     };
 
+    const [filterBy, setFilterBy] = useState<string | undefined>();
+    const [filterValue, setFilterValue] = useState<unknown>();
+    const onFilter = (property: string, value: unknown) => {
+        setFilterBy(property);
+        setFilterValue(value);
+    };
```
Comment (wejendorp) [inline src/web/components/table/KapTable.tsx:41]: If they are always set together, maybe they should just be one state object.
Also we might want to `useCallback` on this?

File: src/web/inspectors/requests/StatusCodeFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                />
+              </ListItemIcon>
+              <ListItemText id={labelId} primary={`Line item ${value + 1}`} />
+            </ListItemButton>
+          </ListItem>
+        );
+      })}
+    </List>
+
+ */
```
Comment (wejendorp) [inline src/web/inspectors/requests/StatusCodeFilter.tsx:121]: Delete?

**Review by wejendorp — APPROVED** [review]

File: src/web/components/table/KapTable.tsx (hunk 1 above)
Comment (andertun) [reply src/web/components/table/KapTable.tsx]: I wrapped it in `useCallback` 👍  76b6eb7b79516e69d013003884ff5a5d897d1455

I prefer keeping them as two separate primitives as it simplifies where they are used. They are only updated in this `onFilter` function, so we are not calling them in pair other places.

File: src/web/inspectors/requests/StatusCodeFilter.tsx (hunk 2 above)
Comment (andertun) [reply src/web/inspectors/requests/StatusCodeFilter.tsx]: 092319c82975480fc4e4bc49f5788e57dc8cee65

**Pull request merged**